src/main/scala/mesosphere/marathon/api/v2/AppInfoEmbedResolver.scala

```diff
@@ -0,0 +1,56 @@
+package mesosphere.marathon.api.v2
+
+import java.util
+
+import mesosphere.marathon.core.appinfo.AppInfo
+import mesosphere.marathon.core.appinfo.AppInfo.Embed
+import org.slf4j.LoggerFactory
+
+/**
+  * Resolves AppInfo.Embed from query parameters.
+  */
+private[v2] object AppInfoEmbedResolver {
+  private[this] val log = LoggerFactory.getLogger(getClass)
+
+  private[this] val EmbedAppsPrefixes = Set("apps.", "app.")
+
+  private[this] val EmbedTasks = "tasks"
+  private[this] val EmbedDeployments = "deployments"
+
+  /* deprecated, use lastTaskFailure, tasks, deployments instead */
+  private[this] val EmbedTasksAndFailures = "failures"
+  private[this] val EmbedLastTaskFailure = "lastTaskFailure"
+  private[this] val EmbedCounts = "counts"
+
+  /**
+    * Converts embed arguments to our internal representation.
+    *
+    * Accepts the arguments with all prefixes or even no prefix at all
+    * to avoid subtle user errors confusing the two.
+    */
+  def resolve(embed: util.Set[String]): Set[Embed] = {
+    def mapEmbedStrings(prefix: String, withoutPrefix: String): Set[AppInfo.Embed] = withoutPrefix match {
```

Comment:
"tasks"=> Tasks, "deployments"=>Deployments should be part of embed. The resolve method makes sense here, since it removes the prefix and creates a set...